src/gennylib/src/PoolManager.cpp

```diff
@@ -38,9 +39,20 @@ auto createPool(const std::string& name,
         poolFactory.setOptions(genny::v1::PoolFactory::kAccessOption, *accessOpts);
     }
 
+    poolFactory.setEncryptionContext(std::move(encryption));
     return poolFactory.makePool();
 }
 
+auto setupEncryption(const std::string& name, const Node& context, bool dryRun) {
+    auto mongoUri = context["Clients"][name]["URI"].to<std::string>();
+    auto encryption = std::make_shared<EncryptionContext>(
+        context["Clients"][name]["EncryptionOptions"], std::move(mongoUri));
+    if (!dryRun) {
+        encryption->setupKeyVault();
```

Comment:
if `setupKeyVault` took a `mongoUri` as a parameter then the EncryptionContext wouldn't need to store the value and then maybe the new Pools struct could be rolled back?